cnxarchive/search.py

```diff
@@ -631,9 +631,48 @@ def _build_search(structured_query, weights):
     sorts.extend(('weight DESC', 'uuid DESC',))
     sorts = ', '.join(sorts)
 
+    # add the columns to get from the database
+    if columns is None:
+        columns = """
+lm.name as title, title_order(lm.name) as "sortTitle",
+lm.uuid as id,
+CASE
+  WHEN lm.portal_type = 'Collection'
+    THEN lm.major_version || '.' || lm.minor_version
+  ELSE lm.major_version || ''
+END AS version,
+language,
+lm.portal_type as "mediaType",
+iso8601(lm.revised) as "pubDate",
+ARRAY(SELECT k.word FROM keywords as k, modulekeywords as mk
+      WHERE mk.module_ident = lm.module_ident
+            AND mk.keywordid = k.keywordid) as keywords,
+ARRAY(SELECT tags.tag FROM tags, moduletags as mt
+      WHERE mt.module_ident = lm.module_ident
+            AND mt.tagid = tags.tagid) as subjects,
+ARRAY(SELECT row_to_json(user_rows) FROM
+      (SELECT username as id,
+              first_name as firstname, last_name as surname,
+              full_name as fullname, title, suffix
+       FROM users
+       WHERE users.username::text = ANY (lm.authors)
+       ) as user_rows) as authors,
+       -- The following are used internally for further sorting and debugging.
+       weight, rank,
+       keys as _keys, '' as matched, '' as fields,
+       ts_headline(ab.html,plainto_tsquery(%(text_terms)s),
+            'ShortWord=5, MinWords=50, MaxWords=60') as abstract,
+       -- until we actually do something with it
+       -- ts_headline(mfti.fulltext, plainto_tsquery(%(text_terms)s),
+       --            'StartSel=<b>, StopSel=</b>, ShortWord=5, MinWords=50,
+       --                MaxWords=60') as headline
+       NULL as headline
+     -- Only retrieve the most recent published modules.
+                  """
+
```

Comment:
yeah. I didn't know the best way to do the search because I was gonna use the search build from cnx-db and just change the columns it was selecting, but selecting the columns was a large part of query.sql in cnx-db. But if the search should change then This can be should back.